### PR TITLE
VZ-4250 enhance OCIR scan job to include scans with trivy and grype

### PR DESCRIPTION
**Pull request by agv426**
# Description

This adds trivy and grype scanning of the images in the specified BOM. It also calls this when checking OCIR scan results so we have the trivy and grype results included alongside the OCIR scan results.

The consolidated report side is a separate sub-task being worked in parallel.

The allowlist support will be looked at in a separate task (VZ-4337)

# Checklist 

As the author of this PR, I have:

- [ ] Checked that I included or updated copyright and license notices in all files that I altered
- [ ] Added or updated unit tests for any new functions I added
- [ ] Added or updated integration tests if appropriate
- [ ] Added or updated acceptance tests if appropriate

Code reviewer, please confirm this PR:

- [ ] Addressed the requirement and meets the acceptance criteria
- [ ] Does not introduce unrelated or spurious changes
- [ ] Does not introduce any unapproved dependency
- [ ] Makes sense and it easy to understand, and/or difficult areas of code are clearly documented so that they can be understood

### CODE REVIEW

**Review by fredtibbitts — APPROVED** [review]

**Review by fredtibbitts — DISMISSED** [review]

File: release/scripts/generate_vulnerability_report.sh [hunk 1]
```diff
@@ -48,33 +47,34 @@ function get_ocir_image_list () {
...
+            -e "s/-trivy-details.csv//" | sort -u
     return 0
 }
 
 function get_file_path () {
     local _image="$1"
     local _scanner="$2"
-    local _suffix="$3"
-    echo "${scan_folder}/${file_prefix}${_image}${_scanner}${_suffix}"
+	local _prefix="$3"
```
Comment (fredtibbitts) [inline release/scripts/generate_vulnerability_report.sh:81]: Minor, spacing is off.

File: release/scripts/generate_vulnerability_report.sh [hunk 2]
```diff
@@ -162,8 +162,9 @@ function process_record () {
 
 function process_csv_details_file () {
     local _image="$1"
-    local _scannersuffix="$2"
-    local _csvdetails=$(get_file_path "${_image}" "${_scannersuffix}" "${file_suffix_csvdetails}")
+	local _prefix="$2"
```
Comment (fredtibbitts) [inline release/scripts/generate_vulnerability_report.sh:171]: Spacing.

**Review by fredtibbitts — COMMENTED** [review]
A couple of minor spacing issues, otherwise looks good to me.

File: release/scripts/generate_vulnerability_report.sh (hunk 2 above)
Comment (agv426) [reply release/scripts/generate_vulnerability_report.sh:171]: fixing, I had edited some files in beyondcompare and pushed them back from there, I think it added tabs there

**Review by fredtibbitts — APPROVED** [review]

**Pull request merged**